Fixed dropdown and pagination

JonathanGuo opened the pull request:
Fixed initialisation of dynamic content in drop down. it had an issue
when rendering dynamic content in drop down.
e.g.

``` HTML
<a class='dropdown-button btn' href='javascript:void(0);'
data-activates='demoDropdown' dropdown data-hover="true">
    {{ dynamic_content }}
</a>
```

it renders nothing initially. I moved $compile inside $timeout to fix the
issue.
## 

Fixed ulClass of pagination. Now the directive accepts object, string.
ulClass is not mandatory.

JonathanGuo committed:
Fixed dropdown and pagination

Fixed initialisation of dynamic content in drop down. it had an issue
when rendering dynamic content in drop down.
e.g.
<a class='dropdown-button btn' href='javascript:void(0);'
data-activates='demoDropdown' dropdown data-hover="true">
    {{ dynamic_content }}
</a>

it renders nothing initially. I put $compile inside $timeout to fix the
issue.

Fixed ulClass of pagination. Now the directive accepts object, string.
ulClass is not mandatory.

webbiesdk commented:
When I checkout your version, I get this error in index.html: 
http://i.imgur.com/odIBFK9.png

From what I understand, it happens because angular is trying to set up a 2-way binding for `ulClass`. But `ulClass` isn't defined on the directive (because there is no ulClass attribute defined on the pagination tag), so angular therefore cannot setup the 2-way binding. 

That is why I suggested to use `"=?"` here: https://github.com/krescruz/angular-materialize/commit/67c439d0726a7d39a023b30c98c38e7486498ed3. 

Don't you have the error in my screenshot when there is no `ulClass` attribute on the pagination tag?

JonathanGuo commented:
Nope. Probably because I used 1.4.3. I am gonna check the older versions then create a new pull request.

JonathanGuo committed:
Removed redundant code of dropdown and made ulClass optional.

Removed redundant code of dropdown by moving $compile into the same
$timeout closure.

The reason of moving $compile into $timeout is for rendering dynamic
content initially.

—
Updated ulClass from ‘=‘ to ‘=?’. Now it’s optional. Tested with 1.2.*,
1.3.* and 1.4.*.

JonathanGuo commented:
Hi @webbiesdk , please check commit a72cbaf8b58abc97ef912539ff73acde28a503c2. I have tested the pagination directive with 1.2._, 1.3._ and 1.4.*.

webbiesdk commented:
Nice work.
